common/modifiedHashes_test.go

```diff
@@ -24,3 +26,58 @@ func TestModifiedHashes_Clone(t *testing.T) {
 	}
 
 }
+
+func TestModifiedHashesSlice_Append(t *testing.T) {
+	t.Parallel()
+
+	mhs := NewModifiedHashesSlice()
+	numHashes := 100
+	hashes := make([][]byte, numHashes)
+	for i := 0; i < numHashes; i++ {
+		hashes[i] = []byte{byte(i)}
+	}
+
+	wg := sync.WaitGroup{}
+	wg.Add(numHashes)
+	for i := 0; i < numHashes; i++ {
+		go func(index int) {
+			mhs.Append([][]byte{hashes[index]})
+			wg.Done()
+		}(i)
+	}
+	wg.Wait()
+
+	for i := 0; i < numHashes; i++ {
+		hashFound := false
+		for _, hash := range mhs.hashes {
+			if bytes.Equal(hashes[i], hash) {
+				hashFound = true
+			}
+		}
+		assert.True(t, hashFound)
+	}
+}
+
+func TestModifiedHashesSlice_Get(t *testing.T) {
+	t.Parallel()
+
+	mhs := NewModifiedHashesSlice()
+	hashes := [][]byte{{1}, {2}, {3}}
+	mhs.Append(hashes)
+	retrievedHashes := mhs.Get()
+	assert.Equal(t, hashes, retrievedHashes)
+	retrievedHashes[0] = []byte{4}
+	newRetrievedHashes := mhs.Get()
+	assert.NotEqual(t, retrievedHashes, newRetrievedHashes)
+}
+
+func TestModifiedHashesSlice_Reset(t *testing.T) {
```

Comment:
add a test with all these operations combined? append, reset, append

Reply:
Removed the Reset() func